<a href="https://colab.research.google.com/github/veronica1908/SALUD_E3_AIII/blob/main/a_Limpieza_transformaci%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Cargar paquetres y librerías

In [1]:
import pandas as pd
import numpy as np

In [60]:
from google.colab import files

#Cargar las bases de datos

In [2]:
# Cargar la base de egresos hospitalarios
df_egresos = pd.read_csv('RETO_df_egresos.csv')

# Cargar la base de usuarios
df_usuarios = pd.read_csv('RETO_df_usuarios.csv')

<ipython-input-2-3846a51f3149>:5: DtypeWarning: Columns (4,6,7,9,10,11,12) have mixed types. Specify dtype option on import or set low_memory=False.
  df_usuarios = pd.read_csv('RETO_df_usuarios.csv')


In [3]:
# Cargar la base de datos de crónicos codificación ISO-8859-1
df_cronicos = pd.read_csv('RETO_df_cronicos.csv', delimiter=';', encoding='ISO-8859-1')

<ipython-input-3-6c1a9848a714>:2: DtypeWarning: Columns (24,25,34,36,39,40,50,52,55,56,58,62,64,65,66,75,107,108,111,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289) have mixed types. Specify dtype option on import or set low_memory=False.
  df_cronicos = pd.read_csv('RETO_df_cronicos.csv', delimiter=';', encoding='ISO-8859-1')


#Exploración inicial de los datos

In [4]:
print("\nPrimeras filas de la base de usuarios:")
print(df_usuarios.head())

print("\nInformación de la base de usuarios:")
print(df_usuarios.info())

# Ordenar las columnas por cantidad de datos nulos de mayor a menor
print("Variables con  datos nulos:")
sorted_null_values = df_usuarios.isnull().sum().sort_values(ascending=False)

# Mostrar los resultados
print(sorted_null_values)


Primeras filas de la base de usuarios:
     MES  YEAR  NRODOC SEXO     FECHA NACIMIENTO  EDAD DEPARTAMENTO MUNICIPIO  \
0  ABRIL  2017  306221    F  1938-09-09 00:00:00   NaN          NaN       NaN   
1  ABRIL  2017  306222    F  1974-07-03 00:00:00   NaN          NaN       NaN   
2  ABRIL  2017  306223    F  1930-08-24 00:00:00   NaN          NaN       NaN   
3  ABRIL  2017  306224    F  1979-12-01 00:00:00   NaN          NaN       NaN   
4  ABRIL  2017  306225    M  1943-06-01 00:00:00   NaN          NaN       NaN   

  BARRIO FECHA INICIO AL PGP FECHA PRIMERA CLASE FUNCIONAL  \
0    NaN                 NaN                           NaN   
1    NaN                 NaN                           NaN   
2    NaN                 NaN                           NaN   
3    NaN                 NaN                           NaN   
4    NaN                 NaN                           NaN   

  PRIMERA CLASE FUNCIONAL FECHA ÚLTIMA CLASE FUNCIONAL ÚLTIMA CLASE FUNCIONAL  \
0                  

In [5]:
# Mostrar las primeras filas
print("Primeras filas de la base de egresos hospitalarios:")
print(df_egresos.head())

# Información general de los dataframes
print("\nInformación de la base de egresos hospitalarios:")
print(df_egresos.info())

# Ordenar las columnas por cantidad de datos nulos de mayor a menor
print("Variables con  datos nulos:")
sorted_null_values = df_egresos.isnull().sum().sort_values(ascending=False)

# Mostrar los resultados
print(sorted_null_values)

Primeras filas de la base de egresos hospitalarios:
   YEAR        MES  NRO ATENCION  NRO INGRESO  SERVICIO HABILITADO COD  \
0  2017       MAYO       3433964      1899917                     -1.0   
1  2017  DICIEMBRE       3720631      2018136                     -1.0   
2  2018    FEBRERO       3912821      2159592                    501.0   
3  2018       MAYO       4023613      2254075                    501.0   
4  2018      JULIO       4080166      2292863                    101.0   

     SERVICIO HABILITADO TIPO IDENTIFICACION  NRODOC         FECHA NACIMIENTO  \
0        Sin Informacion                  CC  308185  1937-07-18 00:00:00.000   
1        Sin Informacion                  CC  308185  1937-07-18 00:00:00.000   
2  Urgencias >= 24 horas                  CC  308185  1937-07-18 00:00:00.000   
3  Urgencias >= 24 horas                  CC  308185  1937-07-18 00:00:00.000   
4        General adultos                  CC  308185  1937-07-18 00:00:00.000   

  REGIMEN AFILIA

In [6]:
print("\nPrimeras filas de la base de crónicos:")
print(df_cronicos.head())

print("\nInformación de la base de crónicos:")
print(df_cronicos.info())

# Ordenar las columnas por cantidad de datos nulos de mayor a menor
print("Variables con  datos nulos:")
sorted_null_values = df_cronicos.isnull().sum().sort_values(ascending=False)

# Mostrar los resultados
print(sorted_null_values)


Primeras filas de la base de crónicos:
   YEAR      MES Fecha diligenciamiento           Hora  Atención  Ingreso  \
0  2017  OCTUBRE              2/10/2017  9:00:20 a. m.    3912499  2629714   
1  2017  OCTUBRE              2/10/2017  9:04:33 a. m.    3942247  2629726   
2  2017  OCTUBRE              2/10/2017  9:11:43 a. m.    3898560  2629751   
3  2017  OCTUBRE              2/10/2017  9:33:34 a. m.    3912599  2629828   
4  2017  OCTUBRE              2/10/2017  9:43:59 a. m.    3906820  2629717   

  Tipo  NRODOC  Peso  Talla  ...  Diagnostico79 NombreDiagnostico79  \
0   CC  311144  69.0  155.0  ...            NaN                 NaN   
1   CC  308418  62.0  155.0  ...            NaN                 NaN   
2   CC  308573  89.0  163.0  ...            NaN                 NaN   
3   CC  307549  88.0  163.0  ...            NaN                 NaN   
4   CC  307044  58.0  158.0  ...            NaN                 NaN   

   Diagnostico80  NombreDiagnostico80  Diagnostico81  NombreDiag

#Preprocesamiento

##Base Usuarios

In [7]:
# Vamos a filtrar la base solo con los usuarios que tengan 60 años o más.
df_usuarios60 = df_usuarios[df_usuarios['EDAD'] >= 60]

print("\nInformación de la base de usuarios:")
print(df_usuarios60.info())


Información de la base de usuarios:
<class 'pandas.core.frame.DataFrame'>
Index: 130595 entries, 24410 to 183910
Data columns (total 16 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   MES                            130595 non-null  object 
 1   YEAR                           130595 non-null  int64  
 2   NRODOC                         130595 non-null  int64  
 3   SEXO                           130595 non-null  object 
 4   FECHA NACIMIENTO               71743 non-null   object 
 5   EDAD                           130595 non-null  float64
 6   DEPARTAMENTO                   71631 non-null   object 
 7   MUNICIPIO                      70954 non-null   object 
 8   BARRIO                         96960 non-null   object 
 9   FECHA INICIO AL PGP            109440 non-null  object 
 10  FECHA PRIMERA CLASE FUNCIONAL  76778 non-null   object 
 11  PRIMERA CLASE FUNCIONAL        78149 non-null   object 

Pasamos de 183911 datos a 130595. Ahora como solo nos interesa la información de los usuarios de acuerdo con su edad, vamos a eliminar las variables que tengan datos nulos y nos quedamos con las que estén completas.

In [8]:
# Eliminar las columnas con datos nulos
df_usuarios60 = df_usuarios60.dropna(axis=1)

# Mostrar las columnas restantes
print(df_usuarios60.columns)

Index(['MES', 'YEAR', 'NRODOC', 'SEXO', 'EDAD', 'CICLO_VITAL', 'QUINQUENIO'], dtype='object')


Pasamos de 16 variables a 7 y con estas trabajaremos en adelante.

##Base Egresos

In [9]:
# Lo primero es filtrar los registros que correspondan al número de documento de los usuarios que fueron filtrados en la base anterior, que corresponden a edad =>60 años
# Filtrar la base de egresos usando la variable NRODOC de la base usuarios
df_egresos60 = df_egresos[df_egresos['NRODOC'].isin(df_usuarios60['NRODOC'])]

# Verificar el nuevo conjunto de datos
print(df_egresos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 5781 entries, 0 to 6375
Data columns (total 66 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   YEAR                               5781 non-null   int64  
 1   MES                                5781 non-null   object 
 2   NRO ATENCION                       5781 non-null   int64  
 3   NRO INGRESO                        5781 non-null   int64  
 4   SERVICIO HABILITADO COD            5781 non-null   float64
 5   SERVICIO HABILITADO                5781 non-null   object 
 6   TIPO IDENTIFICACION                5781 non-null   object 
 7   NRODOC                             5781 non-null   int64  
 8   FECHA NACIMIENTO                   5781 non-null   object 
 9   REGIMEN AFILIACION                 5781 non-null   object 
 10  EPS VALIDADA                       3736 non-null   object 
 11  MODALIDAD CONTRATO                 5781 non-null   object 
 1

Pasó de tener 6376 registros a 5781 que corresponden a los usuarios con edad igual o mayor a los 60 años.

In [10]:
# Verificamos las cantidades de datos nulos por variable
# Ordenar las columnas por cantidad de datos nulos de mayor a menor
print("Variables con  datos nulos:")
sorted_null_values = df_egresos60.isnull().sum().sort_values(ascending=False)

# Mostrar los resultados
print(sorted_null_values)

Variables con  datos nulos:
FECHA POSIBLE ALTA              5646
CAUSA BASICA CAPITULO COD       5294
CAUSA BASICA CATEGORIA COD      5294
CAUSA BASICA MUERTE COD         5292
PERTINENCIA DIAGNOSTICA         4706
                                ... 
SERVICIO EGRESO                    0
TIPO EGRESO                        0
FECHA ALTA MEDICA                  0
MES                                0
PROFESIONAL ESPECIALIDAD GRD       0
Length: 66, dtype: int64


In [11]:
##########################Evaluar si aplicarlo o no, pasaría de tener 66 variables a  tener 52#################
# ELIMINAR LAS VARIABLES (COLUMNAS) CON MÁS DEL 20% DE DATOS NULOS
#Calcular el porcentaje de datos nulos por columna
nulos_por_columna = df_egresos60.isnull().sum() / len(df_egresos60) * 100

# Seleccionar las columnas con más del 20% de datos nulos
columnas_eliminar = nulos_por_columna[nulos_por_columna > 20].index

# Eliminar las columnas seleccionadas del DataFrame
df_egresos60 = df_egresos60.drop(columns=columnas_eliminar)

# Verificar el nuevo conjunto de datos
print(df_egresos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 5781 entries, 0 to 6375
Data columns (total 52 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   YEAR                               5781 non-null   int64  
 1   MES                                5781 non-null   object 
 2   NRO ATENCION                       5781 non-null   int64  
 3   NRO INGRESO                        5781 non-null   int64  
 4   SERVICIO HABILITADO COD            5781 non-null   float64
 5   SERVICIO HABILITADO                5781 non-null   object 
 6   TIPO IDENTIFICACION                5781 non-null   object 
 7   NRODOC                             5781 non-null   int64  
 8   FECHA NACIMIENTO                   5781 non-null   object 
 9   REGIMEN AFILIACION                 5781 non-null   object 
 10  MODALIDAD CONTRATO                 5781 non-null   object 
 11  FUENTE FINANCIACION1               5781 non-null   object 
 1

Pasamos de 63 variables a 52. Ahora vamos a tratar los datos nulos que quedan en estas variables, para ello se eliminan las filas que poseen datos nulos, ya que no representa una gran cantidad para la base.

In [12]:
# Eliminar filas con datos nulos #################DEFINIR SI LO HACEMOS O NO, SOLO SE PIERDE EL 3.2% DE LOS DATOS
df_egresos60 = df_egresos60.dropna()

# Verificar el nuevo conjunto de datos
print(df_egresos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 5595 entries, 0 to 6375
Data columns (total 52 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   YEAR                               5595 non-null   int64  
 1   MES                                5595 non-null   object 
 2   NRO ATENCION                       5595 non-null   int64  
 3   NRO INGRESO                        5595 non-null   int64  
 4   SERVICIO HABILITADO COD            5595 non-null   float64
 5   SERVICIO HABILITADO                5595 non-null   object 
 6   TIPO IDENTIFICACION                5595 non-null   object 
 7   NRODOC                             5595 non-null   int64  
 8   FECHA NACIMIENTO                   5595 non-null   object 
 9   REGIMEN AFILIACION                 5595 non-null   object 
 10  MODALIDAD CONTRATO                 5595 non-null   object 
 11  FUENTE FINANCIACION1               5595 non-null   object 
 1

Se pasa de 5781 registros a 5595, lo que sigue siendo una cantidad representativa de la información, perdiendo solo el 3.2% de los datos.

In [13]:
# Convertir columnas de fecha al tipo datetime
df_egresos60['FECHA NACIMIENTO'] = pd.to_datetime(df_egresos60['FECHA NACIMIENTO'])
df_egresos60['FECHA ALTA MEDICA'] = pd.to_datetime(df_egresos60['FECHA ALTA MEDICA'])
df_egresos60['FECHA SALIDA'] = pd.to_datetime(df_egresos60['FECHA SALIDA'])

# Convertir columnas numéricas a int
columnas_numericas = ['NRO ATENCION', 'NRO INGRESO', 'NRODOC', 'SERVICIO HABILITADO COD']
df_egresos60[columnas_numericas] = df_egresos60[columnas_numericas].astype(int)

# Verificar el nuevo conjunto de datos
print(df_egresos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 5595 entries, 0 to 6375
Data columns (total 52 columns):
 #   Column                             Non-Null Count  Dtype         
---  ------                             --------------  -----         
 0   YEAR                               5595 non-null   int64         
 1   MES                                5595 non-null   object        
 2   NRO ATENCION                       5595 non-null   int64         
 3   NRO INGRESO                        5595 non-null   int64         
 4   SERVICIO HABILITADO COD            5595 non-null   int64         
 5   SERVICIO HABILITADO                5595 non-null   object        
 6   TIPO IDENTIFICACION                5595 non-null   object        
 7   NRODOC                             5595 non-null   int64         
 8   FECHA NACIMIENTO                   5595 non-null   datetime64[ns]
 9   REGIMEN AFILIACION                 5595 non-null   object        
 10  MODALIDAD CONTRATO                 5595 n

<ipython-input-13-69a09fd26ccf>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_egresos60['FECHA NACIMIENTO'] = pd.to_datetime(df_egresos60['FECHA NACIMIENTO'])
<ipython-input-13-69a09fd26ccf>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_egresos60['FECHA ALTA MEDICA'] = pd.to_datetime(df_egresos60['FECHA ALTA MEDICA'])
<ipython-input-13-69a09fd26ccf>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value inste

##Base Crónicos

In [14]:
# Lo primero es filtrar los registros que correspondan al número de documento de los usuarios que fueron filtrados en la base usuarios, que corresponden a edad =>60 años
# Filtrar la base de cronicos usando la variable NRODOC de la base usuarios
df_cronicos60 = df_cronicos[df_cronicos['NRODOC'].isin(df_usuarios60['NRODOC'])]

# Verificar el nuevo conjunto de datos
print(df_cronicos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 35055 entries, 0 to 38716
Columns: 290 entries, YEAR to NombreDiagnostico83
dtypes: float64(36), int64(11), object(243)
memory usage: 77.8+ MB
None


Pasó de tener 38717 registros a 35055 que corresponden a los usuarios con edad igual o mayor a los 60 años.

In [15]:
# Verificamos las cantidades de datos nulos por variable
# Ordenar las columnas por cantidad de datos nulos de mayor a menor
print("Variables con  datos nulos:")
sorted_null_values = df_cronicos60.isnull().sum().sort_values(ascending=False)

# Mostrar los resultados
print(sorted_null_values)

Variables con  datos nulos:
Otras Morbilidades                        35055
NombreDiagnostico82                       35042
Diagnostico82                             35042
NombreDiagnostico83                       35042
Diagnostico83                             35042
                                          ...  
Calificación (Apoyo Monopodal)               0
Tiempo en Segundos (Recorrer 5 metros)        0
Velocidad (m/s)                               0
Calificación Velocidad                       0
YEAR                                          0
Length: 290, dtype: int64


In [16]:
##########################Evaluar si aplicarlo o no, pasaría de tener 290 variables a  tener 72#################
# ELIMINAR LAS VARIABLES (COLUMNAS) CON MÁS DEL 20% DE DATOS NULOS
#Calcular el porcentaje de datos nulos por columna
nulos_por_columna = df_cronicos60.isnull().sum() / len(df_cronicos60) * 100

# Seleccionar las columnas con más del 20% de datos nulos
columnas_eliminar = nulos_por_columna[nulos_por_columna > 20].index

# Eliminar las columnas seleccionadas del DataFrame
df_cronicos60 = df_cronicos60.drop(columns=columnas_eliminar)

# Verificar el nuevo conjunto de datos
print(df_cronicos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 35055 entries, 0 to 38716
Data columns (total 72 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   YEAR                                       35055 non-null  int64  
 1   MES                                        35055 non-null  object 
 2   Fecha diligenciamiento                     35055 non-null  object 
 3   Hora                                       35055 non-null  object 
 4   Atención                                  35055 non-null  int64  
 5   Ingreso                                    35055 non-null  int64  
 6   Tipo                                       35055 non-null  object 
 7   NRODOC                                     35055 non-null  int64  
 8   Peso                                       35055 non-null  float64
 9   Talla                                      35055 non-null  float64
 10  IMC                        

Pasamos de 290 variables a 72. Ahora verificamos las variables de fechas.

In [29]:
# Verificar la forma de los datos de las variables de fechas para aplicar formato
print(df_cronicos60[['Fecha diligenciamiento','Hora','Fecha glicemia','Fecha hemoglobina glicosilada','Fecha HDL',
                     'Fecha colesterol total','Fecha trigliceridos','Fecha creatinina1','Fecha microalbuminuria']])

      Fecha diligenciamiento                Hora Fecha glicemia  \
0                 2017-10-02 2024-04-17 09:00:20            NaT   
1                 2017-10-02 2024-04-17 09:04:33            NaT   
2                 2017-10-02 2024-04-17 09:11:43            NaT   
3                 2017-10-02 2024-04-17 09:33:34            NaT   
4                 2017-10-02 2024-04-17 09:43:59            NaT   
...                      ...                 ...            ...   
38712             2022-05-31 2024-04-17 22:55:31            NaT   
38713             2022-05-31 2024-04-17 22:59:58            NaT   
38714             2022-05-31 2024-04-17 23:13:57            NaT   
38715             2022-05-31 2024-04-17 23:17:35            NaT   
38716             2022-05-31 2024-04-17 23:25:39            NaT   

      Fecha hemoglobina glicosilada Fecha HDL Fecha colesterol total  \
0                               NaT       NaT                    NaT   
1                               NaT       NaT      

In [40]:
# Como las variables 'Fecha glicemia','Fecha hemoglobina glicosilada','Fecha HDL','Fecha colesterol total','Fecha trigliceridos','Fecha creatinina1','Fecha microalbuminuria'
# no contienen realmente datos, se procede a eliminar dichas columnas.

columns_to_drop = ['Fecha glicemia', 'Fecha hemoglobina glicosilada',
                   'Fecha HDL', 'Fecha colesterol total', 'Fecha trigliceridos', 'Fecha creatinina1',
                   'Fecha microalbuminuria']

df_cronicos60.drop(columns=columns_to_drop, inplace=True)

#Quedamos con 65 variables

Ahora vamos a tratar los datos nulos que quedan en estas variables, para ello se eliminan las filas que poseen datos nulos.

In [41]:
# Eliminar filas con datos nulos #################DEFINIR SI LO HACEMOS O NO, SE PIERDE EL 32.7% DE LOS REGISTROS.
df_cronicos60 = df_cronicos60.dropna()

# Verificar el nuevo conjunto de datos
print(df_cronicos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 23536 entries, 0 to 38716
Data columns (total 65 columns):
 #   Column                                     Non-Null Count  Dtype         
---  ------                                     --------------  -----         
 0   YEAR                                       23536 non-null  int64         
 1   MES                                        23536 non-null  object        
 2   Fecha diligenciamiento                     23536 non-null  datetime64[ns]
 3   Hora                                       23536 non-null  datetime64[ns]
 4   Atención                                  23536 non-null  int64         
 5   Ingreso                                    23536 non-null  int64         
 6   Tipo                                       23536 non-null  object        
 7   NRODOC                                     23536 non-null  int64         
 8   Peso                                       23536 non-null  float64       
 9   Talla                 

Se pasa de 35055 registros a 23563, perdiendo el 32.7% de los datos.

In [43]:
#Ajustar formatos de fechas
df_cronicos60['Fecha diligenciamiento'] = pd.to_datetime(df_cronicos60['Fecha diligenciamiento'])
df_cronicos60['Hora'] = pd.to_datetime(df_cronicos60['Hora'])
# Verificar el nuevo conjunto de datos
print(df_cronicos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 23536 entries, 0 to 38716
Data columns (total 65 columns):
 #   Column                                     Non-Null Count  Dtype         
---  ------                                     --------------  -----         
 0   YEAR                                       23536 non-null  int64         
 1   MES                                        23536 non-null  object        
 2   Fecha diligenciamiento                     23536 non-null  datetime64[ns]
 3   Hora                                       23536 non-null  datetime64[ns]
 4   Atención                                  23536 non-null  int64         
 5   Ingreso                                    23536 non-null  int64         
 6   Tipo                                       23536 non-null  object        
 7   NRODOC                                     23536 non-null  int64         
 8   Peso                                       23536 non-null  float64       
 9   Talla                 

In [51]:
#Renombramos algunas columnas que tienen problemas con la escritura

df_cronicos60.rename(columns={'Atención': 'Atencion'}, inplace=True)
df_cronicos60.rename(columns={'Clasificación IMC': 'Clasificacion IMC'}, inplace=True)
df_cronicos60.rename(columns={'Saturación de Oxígeno (%)': 'Saturacion de oxigeno'}, inplace=True)
df_cronicos60.rename(columns={'Perímetro Muslo': 'Perimetro muslo'}, inplace=True)
df_cronicos60.rename(columns={'Perímetro Cintura': 'Perimetro cintura'}, inplace=True)
df_cronicos60.rename(columns={'Presión Arterial Sistólica': 'Presion arterial sistolica'}, inplace=True)
df_cronicos60.rename(columns={'Presión Arterial Diastólica ': 'Presion arterial diastolica'}, inplace=True)
df_cronicos60.rename(columns={'METS -Índice Metabólico': 'METS indice metabolico'}, inplace=True)
df_cronicos60.rename(columns={'VO2 - Máxima Cantidad de Oxígeno': 'VO2 Maxima cantidad de oxigeno'}, inplace=True)
df_cronicos60.rename(columns={'Índice de Fragilidad Groningen': 'Indice de fragilidad groningen'}, inplace=True)
df_cronicos60.rename(columns={'Calificación (Índice de Fragilidad)': 'Calificacion indice de fragilidad'}, inplace=True)
df_cronicos60.rename(columns={'Calificación (Apoyo Monopodal)': 'Calificacion apoyo monopodal'}, inplace=True)
df_cronicos60.rename(columns={'Calificación Velocidad': 'Calificacion velocidad'}, inplace=True)
df_cronicos60.rename(columns={'Tiempo con el diagnóstico1': 'Tiempo con el diagnostico1'}, inplace=True)
df_cronicos60.rename(columns={'Enfermedad Coronaria (En el último año)': 'Enermedad coronaria en el ultimo año'}, inplace=True)
df_cronicos60.rename(columns={'Insuficiencia Cardíaca': 'Insuficiencia cardiaca'}, inplace=True)
df_cronicos60.rename(columns={'Valvulopatía': 'Valvulopatia'}, inplace=True)
df_cronicos60.rename(columns={'Cuantos cigarrillos día': 'Cuantos cigarrillos al dia'}, inplace=True)
df_cronicos60.rename(columns={'Tasa de Filtración Glomerular TFG': 'Tasa de filtracion glomerular TFG'}, inplace=True)
df_cronicos60.rename(columns={'Tasa de Filtración Glomerular TFG2': 'Tasa de filtracion glomerular TFG2'}, inplace=True)
df_cronicos60.rename(columns={'Úlcera de pie diabético': 'Ulcera de pie diabetico'}, inplace=True)
df_cronicos60.rename(columns={'Remisión': 'Remision'}, inplace=True)
df_cronicos60.rename(columns={'Diagnóstico Principal': 'Diagnostico principal'}, inplace=True)
df_cronicos60.rename(columns={'Análisis y conducta a seguir': 'Analisis y conducta a seguir'}, inplace=True)

# Verificamos
print(df_cronicos60.info())

<class 'pandas.core.frame.DataFrame'>
Index: 23536 entries, 0 to 38716
Data columns (total 65 columns):
 #   Column                                     Non-Null Count  Dtype         
---  ------                                     --------------  -----         
 0   YEAR                                       23536 non-null  int64         
 1   MES                                        23536 non-null  object        
 2   Fecha diligenciamiento                     23536 non-null  datetime64[ns]
 3   Hora                                       23536 non-null  datetime64[ns]
 4   Atencion                                   23536 non-null  int64         
 5   Ingreso                                    23536 non-null  int64         
 6   Tipo                                       23536 non-null  object        
 7   NRODOC                                     23536 non-null  int64         
 8   Peso                                       23536 non-null  float64       
 9   Talla                 

In [50]:
#Ahora verificamos las categorías de las variables
for column in df_cronicos60.columns:
    print(f"Valores únicos para {column}:")
    print(df_cronicos60[column].unique())
    print("------------------------------------------------------")

Valores únicos para YEAR:
[2017 2018 2021 2022]
------------------------------------------------------
Valores únicos para MES:
['OCTUBRE' 'NOVIEMBRE' 'DICIEMBRE' 'ENERO' 'FEBRERO' 'MARZO' 'MAYO'
 'JUNIO' 'JULIO' 'AGOSTO' 'SEPTIEMBRE' 'ABRIL']
------------------------------------------------------
Valores únicos para Fecha diligenciamiento:
<DatetimeArray>
['2017-10-02 00:00:00', '2017-10-11 00:00:00', '2017-10-31 00:00:00',
 '2017-10-03 00:00:00', '2017-10-04 00:00:00', '2017-10-05 00:00:00',
 '2017-10-06 00:00:00', '2017-10-09 00:00:00', '2017-10-25 00:00:00',
 '2017-10-13 00:00:00',
 ...
 '2022-05-06 00:00:00', '2022-05-07 00:00:00', '2022-05-08 00:00:00',
 '2022-05-12 00:00:00', '2022-05-14 00:00:00', '2022-05-16 00:00:00',
 '2022-05-17 00:00:00', '2022-05-23 00:00:00', '2022-05-28 00:00:00',
 '2022-05-29 00:00:00']
Length: 760, dtype: datetime64[ns]
------------------------------------------------------
Valores únicos para Hora:
<DatetimeArray>
['2024-04-17 09:00:20', '2024-04-17 

In [48]:
#Unificamos algunas categorías

# Unificar las categorías 'CC' y 'CC ' en Tipo
df_cronicos60['Tipo'] = df_cronicos60['Tipo'].replace('CC ', 'CC')

# Verificar que las categorías se hayan unificado correctamente
print(df_cronicos60['Tipo'].unique())

['CC' 'CE']


In [52]:
# Ajustar los valores 'Frágil' a 'Frágil' en Calificacion indice de fragilidad
df_cronicos60['Calificacion indice de fragilidad'] = df_cronicos60['Calificacion indice de fragilidad'].replace('Frágil', 'Fragil')

# Verificar que las categorías se hayan ajustado correctamente
print(df_cronicos60['Calificacion indice de fragilidad'].unique())

['Fragil' 'Normal']


In [54]:
# Ajustar los valores 'Frágil' a 'Frágil' en Calificacion velocidad
df_cronicos60['Calificacion velocidad'] = df_cronicos60['Calificacion velocidad'].replace('Malo - Frágil', 'Malo - Fragil')

# Verificar que las categorías se hayan ajustado correctamente
print(df_cronicos60['Calificacion velocidad'].unique())

['Aceptable' 'Deficiente' 'Malo - Fragil' 'Bueno']


In [55]:
# Ajustar los valores 'I481 - Fibrilación auricular persistente' en Diagnostico principal
df_cronicos60['Diagnostico principal'] = df_cronicos60['Diagnostico principal'].replace('I481 - Fibrilación auricular persistente', 'I481 - FIBRILACION AURICULAR PERSISTENTE')

# Verificar que las categorías se hayan ajustado correctamente
print(df_cronicos60['Diagnostico principal'].unique())

['I10X - HIPERTENSION ESENCIAL (PRIMARIA)'
 'E118 - DIABETES MELLITUS NO INSULINODEPENDIENTE CON COMPLICACIONES NO ESPECIFICADAS'
 'E109 - DIABETES MELLITUS INSULINODEPENDIENTE SIN MENCION DE COMPLICACION'
 'E103 - DIABETES MELLITUS INSULINODEPENDIENTE CON COMPLICACIONES OFTALMICAS'
 'E119 - DIABETES MELLITUS NO INSULINODEPENDIENTE SIN MENCION DE COMPLICACION'
 'J448 - OTRAS ENFERMEDADES PULMONARES OBSTRUCTIVAS CRONICAS ESPECIFICADAS'
 'I358 - OTROS TRASTORNOS DE LA VALVULA AORTICA'
 'I350 - ESTENOSIS (DE LA VALVULA) AORTICA'
 'E107 - DIABETES MELLITUS INSULINODEPENDIENTE CON COMPLICACIONES MULTIPLES'
 'I694 - SECUELAS DE ENFERMEDAD CEREBROVASCULAR NO ESPECIFICADA COMO HEMORRAGICA U OCLUSIVA'
 'I255 - CARDIOMIOPATIA ISQUEMICA'
 'J449 - ENFERMEDAD PULMONAR OBSTRUCTIVA CRONICA NO ESPECIFICADA'
 'E104 - DIABETES MELLITUS INSULINODEPENDIENTE CON COMPLICACIONES NEUROLOGICAS'
 'G301 - ENFERMEDAD DE ALZHEIMER DE COMIENZO TARDIO'
 'E039 - HIPOTIROIDISMO NO ESPECIFICADO'
 'E102 - DIABETES MELLI

# Exportamos bases procesadas

In [59]:
df_usuarios60.to_csv('df_usuarios60.csv', index=False)
df_egresos60.to_csv('df_egresos60.csv', index=False)
df_cronicos60.to_csv('df_cronicos60.csv', index=False)

In [61]:
files.download('df_usuarios60.csv')
files.download('df_egresos60.csv')
files.download('df_cronicos60.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>